src/PowerShellEditorServices/Session/PowerShellContext.cs

```diff
@@ -31,6 +33,21 @@ namespace Microsoft.PowerShell.EditorServices
     /// </summary>
     public class PowerShellContext : IDisposable, IHostSupportsInteractiveSession
     {
+        private const string DotNetFrameworkDescription = ".NET Framework";
+
+        private static readonly Action<Runspace, ApartmentState> s_runspaceApartmentStateSetter;
+
+        static PowerShellContext()
+        {
+            // PowerShell ApartmentState APIs aren't available in PSStandard, so we need to use reflection
```

Comment:
Should this be in PowerShellStandard?

Reply:
There's no such property in PSCore, and PSStandard is PSCore ∩ Windows PowerShell. So it makes sense it's not in PSStandard.

Reply:
I think it could possibly be in PSStandard but then throw Not Implemented Exception for PSCore.

We shouldn't need to use reflection.

This is an edge case for sure but we should talk about it.

Cc @SteveL-MSFT @JamesWTruher

Reply:
@tylerl0706 If the member has been removed completely, wouldn't that throw a missing member exception at runtime?

Reply:
PSCore6 currently doesn't support setting apartmentstate.  There's currently no plan to add it back.  One of the things I dislike about .NET Std is APIs that exist but throw PlatformNotSupportedException at runtime meaning I can't rely on build time to find api problems for cross platform.

Reply:
> PSCore6 currently doesn't support setting apartmentstate. There's currently no plan to add it back.

Makes sense and I'm definitely not asking for it. More concerned about:

> One of the things I dislike about .NET Std is APIs that exist but throw PlatformNotSupportedException at runtime meaning I can't rely on build time to find api problems for cross platform

I can understand your dislike for that but the alternative is to use reflection which isn't really elegant, right? 

I don't know the .NET Std team's reasoning behind going down that route but I wouldn't be surprised if this was part of that reasoning:

"not having to rely on reflection for scenarios when you have to do extra work on one platform to get the expected behavior on all"

Reply:
In this case, it's the lesser of two evils.  I don't think reflection in itself is necessary a bad thing as long as you use a publicly documented API.

Reply:
I'm personally in favour of compile time checking helping to identify what would otherwise be an unforeseeable runtime crash. But either way, the reflection done here is (1) done once, (2) efficiently compiled to a delegate and (3) called as readably as I could make it -- so hopefully that smooths over the clumsiness of reflection a little bit.

Reply:
That's totally fine. Just wanted to bring it up. If we want to go down this route we should at least have a doc on best practices for these sorts of things.